_test/kind/setup.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+AGENT=$1
+JENKINS_CHART_VERSION=${2:-3.11.10}
+AGENT_PATH="jenkins-agents/${AGENT}"
+SCRIPT_DIR=$(dirname -- "$(readlink -f "${BASH_SOURCE[0]}" || realpath "${BASH_SOURCE[0]}")")
+
+if [[ -d ${AGENT_PATH} ]]
+then
+  # Create KinD cluster and load required container images
+  kind create cluster --config ${SCRIPT_DIR}/kind-config.yaml
```

Comment:
cant decide whether we should use the kind github action to handle all the kind stuff or not. happy for it to be merged as-is, just sharing my current thinking.

Reply:
I briefly thought about this as well but given the one line used here I decided against the kind GH action for now. Definitely open to re-evaluate.